runtime/context.go

```diff
@@ -178,11 +178,17 @@ type serverMetadataKey struct{}
 
 // NewServerMetadataContext creates a new context with ServerMetadata
 func NewServerMetadataContext(ctx context.Context, md ServerMetadata) context.Context {
+	if ctx == nil {
+		ctx = context.Background()
```

Comment:
It's the correct behavior for us to create a background context here? Doesn't this run the risk of silently allowing users to screw up their context to not have a timeout at all?